### PR TITLE
[ROCKETMQ-172]log improvement for rocketmq client

### PR DESCRIPTION
**Pull request by Jaskey**
JIRA: https://issues.apache.org/jira/browse/ROCKETMQ-172?jql=project%20%3D%20ROCKETMQ

This should be a pr which contains many improvement among the client code fragment.

@lizhanhui @shroman @vongosling @zhouxinyu 
Please help pointing out more places where log/message should be improved.

### CODE REVIEW

**Comment by vesense** [comment]
LGTM

File: client/src/main/java/org/apache/rocketmq/client/impl/MQClientAPIImpl.java [hunk 1]
```diff
@@ -577,12 +577,12 @@ public void operationComplete(ResponseFuture responseFuture) {
                     }
                 } else {
                     if (!responseFuture.isSendRequestOK()) {
-                        pullCallback.onException(new MQClientException("send request failed", responseFuture.getCause()));
```
Comment (Jaskey) [inline client/src/main/java/org/apache/rocketmq/client/impl/MQClientAPIImpl.java]: But this is not a sl4j 's logging, it's a exception, so direct string is needed.

File: client/src/main/java/org/apache/rocketmq/client/impl/MQClientAPIImpl.java [hunk 2]
```diff
@@ -577,12 +577,12 @@ public void operationComplete(ResponseFuture responseFuture) {
...
+                        pullCallback.onException(new MQClientException("send request failed to " + addr + ".request: " + request, responseFuture.getCause()));
```
Comment (shroman) [inline client/src/main/java/org/apache/rocketmq/client/impl/MQClientAPIImpl.java]: How about writing it this way?
`new MQClientException("send request [" + request+ "] failed to " + addr, responseFuture.getCause())`

File: client/src/main/java/org/apache/rocketmq/client/impl/MQClientAPIImpl.java [hunk 3]
```diff
@@ -577,12 +577,12 @@ public void operationComplete(ResponseFuture responseFuture) {
...
                     } else if (responseFuture.isTimeout()) {
-                        pullCallback.onException(new MQClientException("wait response timeout " + responseFuture.getTimeoutMillis() + "ms",
+                        pullCallback.onException(new MQClientException("wait response from " + addr + " timeout :" + responseFuture.getTimeoutMillis() + "ms" + " request: " + request,
```
Comment (shroman) [inline client/src/main/java/org/apache/rocketmq/client/impl/MQClientAPIImpl.java]: " request: " -> ", request: "

File: client/src/main/java/org/apache/rocketmq/client/impl/MQClientAPIImpl.java (hunk 2 above)
Comment (shroman) [inline client/src/main/java/org/apache/rocketmq/client/impl/MQClientAPIImpl.java]: Or ".request: " -> ". Request: "

File: remoting/src/main/java/org/apache/rocketmq/remoting/netty/NettyRemotingClient.java [hunk 4]
```diff
@@ -321,6 +321,7 @@ public void updateNameServerAddressList(List<String> addrs) {
 
             if (update) {
                 Collections.shuffle(addrs);
+                log.info("name server address updated. NEW : {} , OLD: {}",addrs,old);
```
Comment (shroman) [inline remoting/src/main/java/org/apache/rocketmq/remoting/netty/NettyRemotingClient.java]: NEW, OLD look ugly to me :)
How about 
```
log.info("name server address updated from [{}] to [{}]", addrs, old);
```
?

File: client/src/main/java/org/apache/rocketmq/client/impl/MQClientAPIImpl.java (hunk 2 above)
Comment (Jaskey) [inline client/src/main/java/org/apache/rocketmq/client/impl/MQClientAPIImpl.java]: I would modify the typo , your NO2. advice

File: remoting/src/main/java/org/apache/rocketmq/remoting/netty/NettyRemotingClient.java (hunk 4 above)
Comment (Jaskey) [inline remoting/src/main/java/org/apache/rocketmq/remoting/netty/NettyRemotingClient.java]: @shroman , but OLD , NEW has been already exists in rocketmq's log. please see ConsumerGroupInfo.java for one sample.

Comment (shroman) [inline remoting/src/main/java/org/apache/rocketmq/remoting/netty/NettyRemotingClient.java]: just a suggestion to make it prettier ;)

**Comment by Jaskey** [comment]
@shroman Please review again

File: remoting/src/main/java/org/apache/rocketmq/remoting/netty/NettyRemotingClient.java (hunk 4 above)
Comment (Jaskey) [inline remoting/src/main/java/org/apache/rocketmq/remoting/netty/NettyRemotingClient.java]: Perhaps we can post another PR for some logging format, currently I would just retain the current way

**Comment by vongosling** [comment]
+1

**Comment by dongeforever** [comment]
LGTM +1

**Comment by dongeforever** [comment]
@Jaskey This PR has been merged

**Comment by Jaskey** [comment]
@dongeforever @vongosling @shroman 

Thank you guys!

**Pull request closed without merging**